### DISCUSSION TITLE
unable to launch macOS client 1.3.3 or 1.3.4 because of missing referenced symbols in the objective-C dylib

### DISCUSSION THREAD

**Discussion by chuchusoft** [rustdesk/rustdesk#10178] [General]
### Bug Description

1.3.3 and 1.3.4 do not launch on my macOS 12.7.6 aarch64 system. 1.3.2 does launch. The problem is due to liblibrustdesk.dylib depending on symbols that do not exist in my system's copy of libobjc.A.dylib.

Here's my Terminal output when I try to launch 1.3.3 or 1.3.4:

> 
> % /Users/---/Applications/RustDesk\ 1.3.4.app/Contents/MacOS/RustDesk
> dyld[90184]: Symbol not found: (_objc_release_x0)
>   Referenced from: '/Users/---/Applications/RustDesk 1.3.4.app/Contents/Frameworks/liblibrustdesk.dylib'
>   Expected in: '/usr/lib/libobjc.A.dylib'
> zsh: abort      /Users/---/Applications/RustDesk\ 1.3.4.app/Contents/MacOS/RustDesk
> 

and here are some of the symbols that liblibrustdesk.dylib tries to link with which my system does not have:

> 
> % nm "/Users/---/Applications/RustDesk 1.3.4.app/Contents/Frameworks/liblibrustdesk.dylib"
> ...
>                  U _objc_release_x0
>                  U _objc_release_x1
>                  U _objc_release_x19
>                  U _objc_release_x20
>                  U _objc_release_x21
>                  U _objc_release_x22
>                  U _objc_release_x23
>                  U _objc_release_x24
>                  U _objc_release_x25
>                  U _objc_release_x26
>                  U _objc_release_x8
>                  U _objc_release_x9
> ...
>                  U _objc_retain_x1
>                  U _objc_retain_x8
> 

if you inspect the 1.3.2 version, it does not try to link with those symbols and thus it launches.

### How to Reproduce

Boot up a Mac with 12.7.6 on it (aarch64). Install RustDesk 1.3.3 or 1.3.4. Try to launch it.

### Expected Behavior

RustDesk launches.

### Operating system(s) on local (controlling) side and remote (controlled) side

macOS 12.7.6 -> (any)

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.3 or 1.3.4 -> (any)

### Screenshots

none

**Comment by rustdesk** [comment]
This break change is introduced for supporting loopback audio, Three choices:

- Upgrade OSX
- Use the x64 version
- Use old arm version

**Comment by rustdesk** [comment]
Could you try out the x64 version?

**Reply by chuchusoft** [reply]
I've tried it. It does not have the missing symbols problem but it does not launch either. Here's the Terminal output:

> % /Users/---/Applications/RustDesk\ 1.3.4\ x64.app/Contents/MacOS/RustDesk
> flutter: launch args: []
> flutter: initializing FFI main
> flutter: _appType:main,info1-id:EE574ACD-19E4-5D8B-92EB-DCFF136BC0FD,dir:/Users/---/Documents
> flutter: _globalFFI init
> flutter: registerEventHandler callback_query_onlines address book peer
> flutter: registerEventHandler load_address_book_peers address book peer
> flutter: registerEventHandler callback_query_onlines group peer
> flutter: registerEventHandler load_group_peers group peer
> flutter: registerEventHandler callback_query_onlines recent peer
> flutter: registerEventHandler load_recent_peers recent peer
> flutter: registerEventHandler callback_query_onlines fav peer
> flutter: registerEventHandler load_fav_peers fav peer
> flutter: registerEventHandler callback_query_onlines discovered peer
> flutter: registerEventHandler load_lan_peers discovered peer
> flutter: _globalFFI init end
> flutter: registerEventHandler native_ui native_ui
> flutter: registerEventHandler check_software_update_finish check_software_update_finish
> % echo $?
> 0

after this output is produced, the executable exits with return code 0. No GUI is shown.

Thanks for looking into this.

**Comment by rustdesk** [comment]
Please test this, https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.3.4-aarch64.dmg

**Reply by chuchusoft** [reply]
Thank you. This build works without problems. It connected to Windows 1.3.4 Rustdesk without problems.

Again, thanks for quickly looking into this and fixing it. Greatly appreciated.
